platform/ios/app/MBXViewController.m

```diff
@@ -227,37 +230,47 @@ @implementation MBXViewController
 
 #pragma mark - Setup & Teardown
 
-+ (void)initialize
-{
-    if (self == [MBXViewController class])
-    {
-        [[NSUserDefaults standardUserDefaults] registerDefaults:@{
-            @"MBXUserTrackingMode": @(MGLUserTrackingModeNone),
-            @"MBXShowsUserLocation": @NO,
-            @"MBXDebug": @NO,
-        }];
-    }
-}
-
 - (void)viewDidLoad
 {
     [super viewDidLoad];
 
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveState:) name:UIApplicationDidEnterBackgroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(restoreState:) name:UIApplicationWillEnterForegroundNotification object:nil];
-    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(saveState:) name:UIApplicationWillTerminateNotification object:nil];
+    // Keep track of current map state and debug preferences,
+    // saving and restoring when the application's state changes.
+    self.currentState =  [MBXStateManager sharedManager].currentState;
 
-    [self restoreState:nil];
+    if (!self.currentState) {
+        // Create a new state with the below default values
+        self.currentState = [[MBXState alloc] init];
+
+        self.mapView.showsUserHeadingIndicator = YES;
+        self.mapView.showsScale = YES;
+        self.zoomLevelOrnamentEnabled = NO;
+        self.frameTimeGraphEnabled = NO;
+        self.debugLoggingEnabled = YES;
+    } else {
+        // Revert to the previously saved state
+        self.mapView.showsUserHeadingIndicator = self.currentState.showsUserHeadingIndicator;
```

Comment:
Seems that in order to avoid bugs you may want to call `[self restoreMapState:nil]` instead.